Editor/UI/TreeViews/TreeViewSelection.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.ProjectAuditor.Editor
+{
+    public class TreeViewSelection
+    {
+        public List<string> groups;
+        public List<string> selection;
+
+        public TreeViewSelection()
+        {
+            groups = new List<string>();
+            selection = new List<string>();
+        }
+
+        public TreeViewSelection(TreeViewSelection threadSelection)
+        {
+            groups = new List<string>();
+            selection = new List<string>();
+
+            Set(threadSelection);
+        }
+
+        public void SetAll(string[] names)
+        {
+            groups.Clear();
+             TreeItemIdentifier allIdentifier = new TreeItemIdentifier("All",TreeItemIdentifier.kAll);
+            groups.Add(allIdentifier.nameWithIndex);
+            
+            selection.Clear();
+            foreach (string nameWithIndex in names)
+            {
+                if (nameWithIndex != allIdentifier.nameWithIndex)
+                {
+                    var identifier = new TreeItemIdentifier(nameWithIndex);
+                    if (identifier.index != TreeItemIdentifier.kAll)
+                    {
+                        selection.Add(nameWithIndex);
+                    }
+                }
+            }
+        }
+
+        public void Set(string name)
+        {
+            groups.Clear();
+            selection.Clear();
+            selection.Add(name);
+        }
+
+        public void SetGroup(string groupName)
+        {
+            groups.Clear();
+            selection.Clear();
+
+            TreeItemIdentifier allTreeViewSelection = new TreeItemIdentifier(groupName,TreeItemIdentifier.kAll);
+            groups.Add(allTreeViewSelection.nameWithIndex);
+        }
+
+        public void Set(TreeViewSelection threadSelection)

```

Comment:
threadSelection needs to be renamed (to treeViewSelection?)

Reply:
Done. I'll push this to the head of this branch and try to figure out how to update this pull request with the changes.